src/ivq.tsx

```diff
@@ -1,30 +1,69 @@
 import {h} from 'preact';
-import { IvqConfig } from "./types/IvqConfig";
+import {QuizLoader} from './providers/quiz-loader';
+import {IvqConfig} from './types/IvqConfig';
 
-// @ts-ignore
 export class Ivq extends KalturaPlayer.core.BasePlugin {
   private _player: KalturaPlayerTypes.Player;
+  private _resolveReadyPromise = () => {};
+  private _readyPromise: Promise<void>;
 
-  static defaultConfig: IvqConfig = {
-
-  };
+  static defaultConfig: IvqConfig = {};
 
   constructor(name: string, player: any, config: IvqConfig) {
     super(name, player, config);
     this._player = player;
+    this._readyPromise = this._makeReadyPromise();
   }
 
   get ready() {
-    return Promise.resolve();
+    return this._readyPromise;
   }
 
-  loadMedia(): void {}
+  loadMedia(): void {
+    const kalturaCuePointService: any = this._player.getService('kalturaCuepoints');
+    this._getQuestionsAndAnswers(kalturaCuePointService);
+    this._getQuiz();
+  }
 
   static isValid(): boolean {
     return true;
   }
 
-  reset(): void {}
+  reset(): void {
+    this._readyPromise = this._makeReadyPromise();
+  }
 
   destroy(): void {}
+
+  private _makeReadyPromise = () => {
+    return new Promise<void>(res => {
+      this._resolveReadyPromise = res;
+    });
+  };
+
+  private _getQuestionsAndAnswers(kalturaCuePointService: any) {
+    kalturaCuePointService?.registerTypes([kalturaCuePointService.CuepointType.QUIZ]);
+  }
+
+  private _getQuiz() {
+    this._player.provider
+      .doRequest([{loader: QuizLoader, params: {entryId: this._player.sources.id}}])
+      .then((data: Map<string, any>) => {
+        if (data && data.has(QuizLoader.id)) {
+          const quizLoader = data.get(QuizLoader.id);
+          const quizData = quizLoader?.response?.quiz;
+          const quizAnswers = quizLoader?.response?.quizAnswers;
+          if (!quizData) {
+            this.logger.warn('quiz data absent');
+          } else {
+            // TODO: use quiz data
+          }
+          this._resolveReadyPromise();
+        }
+      })
+      .catch((e: any) => {
+        this.logger.error(e);
```

Comment:
Please check in other repos but I think error is saved for playback severe errors.
Maybe warn is enough